playground/test_playground.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Fail on all errors
+set -e
+# Fail on expand of unset variables
+set -u
+
+# Setup parameters
+HOST=localhost:8080
+WETH_ADDRESS="0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"  # WETH token
+SELL_TOKEN=$WETH_ADDRESS
+BUY_TOKEN="0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"  # USDC token
+AMOUNT="1000000000000000000"  # 1 ETH
+SLIPPAGE=2  # 2%
+PRIVATE_KEY="0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6"
+COW_SETTLMENT_CONTRACT="0x9008D19f58AAbD9eD0D60971565AA8510560ab41"
+COW_VAULT_RELAYER_CONTRACT="0xC92E8bdf79f0507f65a392b0ab4667716BFE0110"
+APPDATA='{"version":"1.3.0","metadata":{}}'
+MAXUINT256="115792089237316195423570985008687907853269984665640564039457584007913129639935"
+
+
+# Run test flow
+echo "Using private key:" $PRIVATE_KEY
+receiver=$(cast wallet address $PRIVATE_KEY)
+
+# Calculate AppData hash
+app_data_hash=$(cast keccak $APPDATA)
+
+# Deposit WETH
+echo "Wrapping some ETH"
+docker exec playground-chain-1 cast send --private-key $PRIVATE_KEY --value 3ether $WETH_ADDRESS > /dev/null
+
+echo "Setting WETH allowance"
+docker exec playground-chain-1 cast send --private-key $PRIVATE_KEY $WETH_ADDRESS "approve(address, uint)" $COW_VAULT_RELAYER_CONTRACT $MAXUINT256 > /dev/null
+
+echo "Request price qoute for buying USDC for WETH"
+quote_response=$( curl --fail-with-body -s -X 'POST' \
+  "http://$HOST/api/v1/quote" \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "sellToken": "'$SELL_TOKEN'",
+  "buyToken": "'$BUY_TOKEN'",
+  "from": "'$receiver'",
+  "receiver": "'$receiver'",
+  "sellTokenBalance": "erc20",
+  "buyTokenBalance": "erc20",
+  "priceQuality": "fast",
+  "signingScheme": "eip712",
+  "onchainOrder": false,
+  "partiallyFillable": false,
+  "kind": "sell",
+  "sellAmountBeforeFee": "'$AMOUNT'"
+}')
+
+buyAmount=$(jq -r --args '.quote.buyAmount' <<< "${quote_response}")
+feeAmount=$(jq -r --args '.quote.feeAmount' <<< "${quote_response}")
+validTo=$(($(date +%s) + 120)) # validity time: now + 2 minutes
+sellAmount=$((AMOUNT - feeAmount))
+
+# Apply slippage
+buy_token_decimals=$(docker exec playground-chain-1 cast call $BUY_TOKEN "decimals()(uint8)") # get decimal places for buy token
+buyAmount=$((buyAmount / (10**buy_token_decimals) )) # discard decimal places
+buyAmount=$((buyAmount - buyAmount * $SLIPPAGE / 100 )) # apply slippage
+buyAmount=$((buyAmount * (10**buy_token_decimals) )) # add decimal places back
+
+
+# Prepare EIP712 message
+eip712_message=$(jq -r --args '
+  .quote|=(.appData="'$app_data_hash'") | 
+  del(.quote.appDataHash) | 
```

Comment:
We probably don't have to delete `appDataHash` here. I think `cast wallet sign` only uses the values of the fields defined in the EIP712 signature description to compute the hash.